Mage/src/main/java/mage/game/events/DiscardedCardsEvent.java

```diff
@@ -0,0 +1,24 @@
+package mage.game.events;
+
+import mage.abilities.Ability;
+import mage.cards.Cards;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author htrajan
+ */
+public class DiscardedCardsEvent extends GameEvent {
+
+    private final Cards discardedCards;
+
+    public DiscardedCardsEvent(UUID targetId, Ability source, UUID playerId, int amount, Cards discardedCards) {
```

Comment:
TargetId is not used so you can remove it from params. 

Also you must copy discardedCards as new list on return (its potentially bugged cause one event have same list, but it can be accessable from many effects and some code can change it by error, e.g. list.removeIf).

Reply:
fixed